Tests/ScanControllerTests.cs

```diff
@@ -0,0 +1,63 @@
+using AutoFixture;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Turnup.Controllers;
+using Turnup.Entities;
+using Turnup.Services;
+using Turnup.Services.ScanService;
+
+namespace Tests;
+
+
+public class ScanControllerTests
+{
+  private Mock<IScanService> _scanService;
+  private Fixture _fixture;
+  private ScanController _controller;
+
+  public ScanControllerTests()
+  {
+    _fixture = new Fixture();
+    _scanService = new Mock<IScanService>();
+  }
+
+  [Fact]
+  public async Task GetListOfProductsByEstablishmentCode()
+  {
+    var establishment= _fixture.Create<Establishment>();
+    var sr = new ServiceResponse<List<Product>>()
+    {
+      Data = establishment.Products
+    };
+    _scanService.Setup(r => r.GetEstablishmentProducts(establishment.EstablishmentCode))
+      .ReturnsAsync(sr);
+    _controller = new ScanController(_scanService.Object);
+
+    var result = await _controller.GetEstablishment(establishment.EstablishmentCode);
+
+    var obj = result.Result as ObjectResult;
+    
+    Assert.Equal(200, obj.StatusCode);
+
```

Comment:
With a test called `GetListOfProductsByEstablishmentCode` I would expect to see some assertions against a list of products being returned. Checking the `200` status code makes sense, but it's not fully testing the `GetEstablishment` controller action yet

Reply:
These are now refactored into 3 test, check for 200, 404 and list of products